schematic/lepton-schematic.scm

```diff
@@ -0,0 +1,379 @@
+   void
+   (dynamic-func "g_init_window" libleptongui)
+   '()))
+
+(define init-select
+  (pointer->procedure
+   void
+   (dynamic-func "g_init_select" libleptongui)
+   '()))
+
+(define init-hook
+  (pointer->procedure
+   void
+   (dynamic-func "g_init_hook" libleptongui)
+   '()))
+
+(define init-action
+  (pointer->procedure
+   void
+   (dynamic-func "g_init_action" libleptongui)
+   '()))
+
+(define init-attrib
+  (pointer->procedure
+   void
+   (dynamic-func "g_init_attrib" libleptongui)
+   '()))
+
+(define init-keys
+  (pointer->procedure
+   void
+   (dynamic-func "g_init_keys" libleptongui)
+   '()))
+
+(define init-builtins
+  (pointer->procedure
+   void
+   (dynamic-func "g_init_builtins" libleptongui)
+   '()))
+
+(define init-util
+  (pointer->procedure
+   void
+   (dynamic-func "g_init_util" libleptongui)
+   '()))
+
+(define init-undo
+  (pointer->procedure
+   void
+   (dynamic-func "scheme_init_undo" libleptongui)
+   '()))
+
+(define init-buffers
+  (pointer->procedure
+   void
+   (dynamic-func "o_buffer_init" libleptongui)
+   '()))
+
+(define init-color
+  (pointer->procedure
+   void
+   (dynamic-func "x_color_init" libleptongui)
+   '()))
+
+(define init-undo*
+  (pointer->procedure
+   void
+   (dynamic-func "o_undo_init" libleptongui)
+   '()))
+
+(define set-quiet-mode!
+  (pointer->procedure
+   void
+   (dynamic-func "set_quiet_mode" libleptongui)
+   '()))
+
+(define set-verbose-mode!
+  (pointer->procedure
+   void
+   (dynamic-func "set_verbose_mode" libleptongui)
+   '()))
+
+
+;;; Localization.
+(define %textdomain "lepton-schematic")
+(bindtextdomain %textdomain "@localedir@")
+(textdomain %textdomain)
+(bind-textdomain-codeset %textdomain "UTF-8")
+(setlocale LC_ALL "")
+(setlocale LC_NUMERIC "C")
+
+(define (G_ msg) (gettext msg %textdomain))
+
+;;; Precompilation.
+(define (precompile-mode)
+  (getenv "LEPTON_SCM_PRECOMPILE"))
+
+(define (precompile-prepare)
+  (setenv "GUILE_AUTO_COMPILE" "0"))
+
+;;; Add Lepton compiled path to Guile compiled paths env var.
+(define (set-guile-compiled-path)
+  (set! %load-compiled-path (cons "@ccachedir@"
+                                  %load-compiled-path)))
+(define (register-guile-funcs)
+  (register-funcs)
+  (init-window)
+  (init-select)
+  (init-hook)
+  (init-action)
+  (init-attrib)
+  (init-keys)
+  (init-builtins)
+  (init-util)
+  (init-undo))
+
+(define (precompile-run)
+  (let ((script (getenv "LEPTON_SCM_PRECOMPILE_SCRIPT")))
+    (if script
+        (begin (register-guile-funcs)
+               ;; Actually load the script.
+               (primitive-load script)
+               0)
+        1)))
+
+(primitive-eval '(use-modules (lepton log)
+                              (lepton version)))
+
+
+(define (process-error-stack stack key args)
+  (match args
+    ((subr message message-args rest)
+     ;; Capture long error message (including possible backtrace).
+     (let ((long-message
+            (with-output-to-string
+              (λ ()
```

Comment:
Excuse me, it's a bit much. As if crazy names like
`make-picture/vector` is not enough. :-(

`$ cat lambda-facepalm.scm`
```scheme
( define λ—это_стильно,модно,молодёжно! "facepalm" )
( define π-ζδϵτσ "excuse me my Greek" )

( define ( функция-шмункция )
  ( format #t "функция-шмункция: ~a~%" функция-шмункция )
  ( format #t "вы будете смеяться, но и это тоже работает.~%" )
  ( format #t "λ—это_стильно,модно,молодёжно! [~a]~%"
               λ—это_стильно,модно,молодёжно! )
  ( format #t "π-ζδϵτσ [~a]~%" π-ζδϵτσ )
)

( функция-шмункция )
```

```
$ guile lambda-facepalm.scm
функция-шмункция: #<procedure функция-шмункция ()>
вы будете смеяться, но и это тоже работает.
λ—это_стильно,модно,молодёжно! [facepalm]
π-ζδϵτσ [excuse me my Greek]
$ echo $?
0
```

Reply:
Oh, come on :-) λ macro is available in guile core since 1.9.2, included by popular request. Many people like it, it looks prettier and the code is shorter. Emacs has a convenient shortcut for insertion it.  You see, I do not use any other non-ASCII letters in the code.

Reply:
> included by popular request

...received from kindergartens, fair enough. :-) They just
don't know there are even more fancy-pancy looking letters, `Ы`, for example.
And every technically inclined young person is well aware that code with
embedded anime pictures is faster and more reliable: this vital feature
is already in active development, I'm sure.
It is evident that replacing `lambda` with `λ` has zero practical benefits.
The only result we'll get is damaged reputation.
If at all possible, please let's stick to ASCII. Thank you.

Reply:
In such a case, Guile, Guix, and other Guile-based projects already have such a bad reputation.  The next step would be to avoid using all such programs.  Don't you know that `lambda` (I'm avoiding using of that horrible UTF-8 letter, probably we have to disable UTF-8 in our project at all) is about *lambda-calculations*?  (Now, after thinking of it a bit more I would prohibit TeX and mathematics completely in Lepton as well, if I could.) You probably aware of some `Ы`-calculations and some programs that we could use instead of Guile?  Let's do it, I'm all for it :-)

Jokes aside (while they're sometimes sad), well OK, apart from this issue, if I'll use just plain `lambda`, are there any other issues in the code that prevent merging it?

Reply:
Please, don't get me wrong. I'm trying to point out that this
particular innovation at this point in time has zero practical
benefits and might bring negative impact. That's all.

I'm not against Unicode or Guile or whatever else the (unfounded)
generalization was made about. The problem lies on a completely
different plane.

Imagine users watching our project. It's evident that there is *huge*
amount of work to be done (suffice it to say, we do not have decent
documentation). In this situation, developers who busy with frills,
bells and whistles couldn't make a great impression. I, for one,
would certainly be alarmed by seeing that.

I've tested the recent changes, configuration is loaded correctly.
I think we could merge it. Thank you.

Reply:
It's not about bells and whistles. It's just convenient and quick and I used this in new code only. Fair enough, let's stop this.
Thank you for the review!